tests/data/ref_dimensions.yaml

```diff
@@ -15,9 +15,78 @@ NXarchive(NXobject):
     my_dim_with_numpy_3d(NX_NUMBER):
       dim: (2, 4, 5)
     my_dim_with_doc_and_dim_attr(NX_NUMBER):
-      dimensions:
+      dim:
+        1:
+          value: 2
+          ref: s
+        3:
+          value: 4
+          ref: m
```

Comment:
FYI, you may follow this PR #24 to resolve the conundrum of `dimension` and `dim` elements. We came to a common approach for them. You may want to extend or design on top of it.

Otherwise, the new design may bring unnecessary modifications in `yaml` and `nxdl` file and will be confusing for others.

Reply:
#24 was merged in already, so already when the feature branch here I started working on, I was building on it